expo/expo
# [versioning][android] rewrite expoview versioning to ts

# Why

getting it on par with iOS versioning

# How

- reuse existing abstractions from iOS versioning
- mostly keep the same logic as the original scripts(I'll add more info in the comments)
- I left some parts that are commented out, I plan to remove them, just left them for the review

# Test Plan

- comment out everything except expoview copy step in https://github.com/expo/expo/blob/main/tools/src/versioning/android/index.ts#L633
- Run `et add-sdk --platform android --sdkVersion 46.0.0`
- commit changes
- switch back to old shell script
- delete versioned code and run add-sdk again
- verify no changes

Note: It was tested on linux

# Checklist
